test/dlc_tests/sanity/quick_checks/test_sanity_fixture.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import os
+import sys
+import re
+
+import pytest
+
+from test.test_utils import PR_ONLY_REASON, get_repository_local_path, is_pr_context
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+LOGGER.addHandler(logging.StreamHandler(sys.stderr))
+
+
+@pytest.mark.quick_checks
+@pytest.mark.skipif(not is_pr_context(), reason=PR_ONLY_REASON)
+@pytest.mark.model("N/A")
+def test_sanity_fixture():
+    """
+    Checks that each sanity test that run within PR or MAINLINE contexts
+    under test/dlc_tests/sanity/ directory contains either
+    `security_sanity` or `functionality_sanity`, not both.
+
+    This test assumes that each test method declare the fixtures
+    using marker pattern `@pytest.mark.usefixtures()` for regex matching.
+    """
+    repository_path = os.getenv("CODEBUILD_SRC_DIR")
+    if not repository_path:
+        repository_path = get_repository_local_path()
+
+    # Look only at test files within the sanity test directory
+    sanity_test_path = os.path.join(repository_path, "test", "dlc_tests", "sanity")
+    LOGGER.debug(f"Test directory: {sanity_test_path}")
+
+    fixture_pattern = r"@pytest.mark.usefixtures\("
+    test_method_pattern = r"def (test_(.*))\("
+    test_fixtures = []
+
+    # Tests that do not run in PR or MAINLINE contexts do not need to have
+    # `security_sanity` or `functionality_sanity` fixtures
+    non_pr_mainline_tests = ["test_deep_canary_integration"]
+
+    # Navigate through files and look at test files at the top level test/dlc_tests/sanity/
+    for item in os.listdir(sanity_test_path):
```

Comment:
there is a simpler way to do this thru pytest api without having to parse thru individual files